src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1109,6 +1143,10 @@ public void setExternalFileAttributes(ZipEntry ze, int externalFileAttributes) {
     // Represents the resolved name and position of a CEN record
     static record EntryPos(String name, int pos) {}
 
+    // Implementation note: This class must be thread-safe.
```

Comment:
I think you mean "This class is thread safe".

Reply:
I was trying to phrase it as text that provides guidance to future maintainers of this class. That's why the use of "must" and the additional text explaining why the "must". But your proposed simpler version is fine too. I have updated the PR to use this text and removed the latter part of the comment.